txrelayer/txrelayer.go

```diff
@@ -61,9 +64,9 @@ type TxRelayerImpl struct {
 
 func NewTxRelayer(opts ...TxRelayerOption) (TxRelayer, error) {
 	t := &TxRelayerImpl{
-		ipAddress:      DefaultRPCAddress,
-		receiptTimeout: 50 * time.Millisecond,
-		numRetries:     defaultNumRetries,
+		ipAddress:        DefaultRPCAddress,
```

Comment:
In the constructor, after applying all the options, calculate `receiptsPollFreq` programmatically. Something like this:

```go
	// Calculate receiptsPollFreq based on receiptsTimeout
	receiptsPollFreq := DefaultPollFreq
	if receiptsTimeout >= time.Second {
		receiptsPollFreq = time.Second
	} else if t.receiptsTimeout >= time.Minute {
		receiptsPollFreq = 2 * time.Second
	} else if t.receiptsTimeout >= time.Millisecond {
		receiptsPollFreq = 500 * time.Millisecond
	}

	if receiptsPollFreq > t.receiptsTimeout {
		// if someone decides to configure a small receipts timeout (in ms) 
		// receiptsPollFreq should be less than receiptsTimeout
		receiptsPollFreq = t.receiptsTimeout / 2
	}
```

There is no benefit in putting poll frequency extremely low (in ms), since block time will never be that low.